vm/vm.go

```diff
@@ -250,7 +250,12 @@ func (vm *VM) Run(program *Program, env interface{}) interface{} {
 			call := vm.constant().(Call)
 			in := make([]reflect.Value, call.Size)
 			for i := call.Size - 1; i >= 0; i-- {
-				in[i] = reflect.ValueOf(vm.pop())
+				popped := vm.pop()
+				if popped == nil {
+					in[i] = reflect.ValueOf(&popped).Elem()
```

Comment:
IN this branch we already know what parameter is nil, but what kind of type should be pass to function?

Reply:
Without the trick reflect.Call failes as reflect.ValueOf(nil) produces zero reflect.Value struct (that is as specified in the documentation). However, reflect.Call does not allow zero reflect.Value as parameter. Therfore we use the trick as follows:
```
var x interface{} = nil
v := reflect.ValueOf(*x).Elem()
```
The v reflect.Value is NOT a zero value and is therfore allowed in reflect.Call. We can still recover the untyped nil by using v.Interface().

The trick does not influence typed nil:
```
var y *int = nil
fmt.Println(reflect.ValueOf(y).Interface() == reflect.ValueOf(*y).Elem().Interface())
```

See also https://play.golang.org/p/Hz3mW6IrGli

Reply:
There was still problem with typed nil, the latest commit fixes it.
https://play.golang.org/p/MyAztoZaqhx

Reply:
What about doing this so: 
```go
				param := vm.pop()
				if param == nil {
					param = &param
				}
				in[i] = reflect.ValueOf(param)
```

Reply:
BTW, this is `false`:
```go
	var a interface{} = nil
	var b interface{} = nil
	z := &b
	fmt.Printf("%T(%#v) %T(%#v) is %v\n", a, b, a, b, a == z)
	// <nil>(<nil>) <nil>(<nil>) is false 
```
https://play.golang.org/p/WYhz019HAku

Reply:
The reflect.Value object in this case would no longer represent nil but a pointer to nil, it would not fix our problem:
https://play.golang.org/p/bcRDIrnz_rN

Reply:
Pointer to nil is not nil so your observation is correct. Thats what the .Elem() is for in the code, we dereference pointer to nil, so we get nil. It should be equivalent:
```
var x interface{} = nil
v1 := reflect.ValueOf(&x).Elem()
v2 := reflect.ValueOf(x)
```
but is not, v1 and v2 are different thing. Thats why it is a GO-tcha and we have the trick to work around this.